### PR TITLE
Remix: use zustand store for env instead of window

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

We're currently relying on a hacky solution for the env variables in the client portion of Remix, injecting them into the `window` object, and this can lead to issues in our components as I found out while developing https://github.com/concrete-utopia/utopia/pull/5069.

**Fix:**

Use the zustand store (non persisted) instead, so we don't rely on `dangerouslySetInnerHTML` and still can access all the env vars. They are initialized at the mount of the root, as it was before when checking `window.ENV`.

### CODE REVIEW

File: utopia-remix/app/components/projectActionContextMenu.tsx [hunk 1]
```diff
@@ -102,10 +104,7 @@ export const ProjectActionsMenu = React.memo(
             'separator',
             {
               text: 'Copy Link',
-              onClick: (selectedProject) => {
-                window.navigator.clipboard.writeText(projectEditorLink(selectedProject.proj_id))
-                // TODO notification toast
-              },
+              onClick: (selectedProject) => copyProjectLink(selectedProject.proj_id),
```
Comment (Rheeseyb) [inline utopia-remix/app/components/projectActionContextMenu.tsx:107]: Oops!

Comment (ruggi) [reply utopia-remix/app/components/projectActionContextMenu.tsx:107]: what am I not seeing here? :D

File: utopia-remix/app/util/use-copy-project-link.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
```
Comment (Rheeseyb) [inline utopia-remix/app/util/use-copy-project-link.tsx:1]: This is meant to be part of #5069 too

File: utopia-remix/app/root.tsx [hunk 3]
```diff
@@ -50,6 +46,11 @@ export const headers: HeadersFunction = () => ({
 export default function App() {
   const data = useLoaderData<typeof loader>()
   const isDarkMode = useIsDarkMode()
+  const setEnv = useProjectsStore((store) => store.setEnv)
```
[the remark this answers is not in the document]
Comment (ruggi) [reply utopia-remix/app/root.tsx]: ha, makes sense. I did a bit of type gymnastics and ended up with [`0d72593` (#5075)](https://github.com/concrete-utopia/utopia/pull/5075/commits/0d72593709186ec94d79dee45502415270c847da), which is based on https://docs.pmnd.rs/zustand/guides/initialize-state-with-props . How does it look?

**Review by balazsbajorics — APPROVED** [review]
looks good!

**Review by bkrmendy — APPROVED** [review]

**Pull request merged**